lib/widgets/pin_code_widget.dart

```diff
@@ -22,13 +24,38 @@ class PinCodeWidgetState extends State<PinCodeWidget> {
   String _enteredPinCode;
   String _errorMessage;
 
+  LocalAuthenticationService _localAuthService;
+  bool _hasBiometricsSet = false;
+  String _availableBiometric;
+
   @override
   initState() {
     super.initState();
+    if (widget.onFingerprintEntered != null) {
+      _availableBiometric = "";
+      _initLocalAuthService();
+    }
     _enteredPinCode = "";
     _errorMessage = "";
   }
 
+  Future _initLocalAuthService() async {
+    _localAuthService = LocalAuthenticationService(context);
```

Comment:
`LocalAuthenticationService` should be exposed only via ServiceInjector and all the biometric authentication logic, including the use of this service, should be inside the bloc that may expose a specific action for that.